admin-dev/themes/new-theme/js/components/pagination/dynamic-grid-paginator.js

```diff
@@ -0,0 +1,254 @@
+    } else {
+      this.updatePaginatorForMiddlePage();
+    }
+    this.gridRenderer.render(data);
+  }
+
+  /**
+   * @private
+   */
+  updatePaginatorForFirstPage() {
+    this.toggleFirstPageAvailability(false);
+    this.togglePreviousPageAvailability(false);
+    this.toggleNextPageAvailability(true);
+    this.toggleLastPageAvailability(true);
+  }
+
+  /**
+   * @private
+   */
+  updatePaginatorForLastPage() {
+    this.toggleNextPageAvailability(false);
+    this.toggleLastPageAvailability(false);
+    this.toggleFirstPageAvailability(true);
+    this.togglePreviousPageAvailability(true);
+  }
+
+  /**
+   * @private
+   */
+  updatePaginatorForMiddlePage() {
+    this.toggleFirstPageAvailability(true);
+    this.togglePreviousPageAvailability(true);
+    this.toggleNextPageAvailability(true);
+    this.toggleLastPageAvailability(true);
+  }
+
+  /**
+   * @param {Number} page
+   *
+   * @private
+   */
+  refreshButtonsData(page) {
+    this.$paginationContainer.find(this.selectorsMap.nextPageBtn).data('page', page + 1);
+    this.$paginationContainer.find(this.selectorsMap.previousPageBtn).data('page', page - 1);
+    this.$paginationContainer.find(this.selectorsMap.lastPageBtn).data('page', this.pagesCount);
+  }
+
+  /**
+   * @param {Boolean} enable
+   *
+   * @private
+   */
+  toggleFirstPageAvailability(enable) {
+    this.toggleTargetAvailability(this.$paginationContainer.find(this.selectorsMap.firstPageItem), enable);
+  }
+
+  /**
+   * @param {Boolean} enable
+   *
+   * @private
+   */
+  toggleLastPageAvailability(enable) {
+    this.toggleTargetAvailability(this.$paginationContainer.find(this.selectorsMap.lastPageItem), enable);
+  }
+
+  /**
+   * @param {Boolean} enable
+   *
+   * @private
+   */
+  togglePreviousPageAvailability(enable) {
+    this.toggleTargetAvailability(this.$paginationContainer.find(this.selectorsMap.previousPageItem), enable);
+  }
+
+  /**
+   * @param {boolean} enable
+   *
+   * @private
+   */
+  toggleNextPageAvailability(enable) {
+    this.toggleTargetAvailability(this.$paginationContainer.find(this.selectorsMap.nextPageItem), enable);
+  }
+
+  /**
+   * @param {Object} target
+   * @param {Boolean} enable
+   *
+   * @private
+   */
+  toggleTargetAvailability(target, enable) {
+    if (enable) {
+      target.removeClass('disabled');
+    } else {
+      target.addClass('disabled');
+    }
+  }
+
+  /**
+   * @param {Number} total
+   *
+   * @private
+   */
+  countPages(total) {
+    this.pagesCount = Math.ceil(total / this.getLimit());
+    const lastPageItem = this.$paginationContainer.find(this.selectorsMap.lastPageBtn);
+    lastPageItem.data('page', this.pagesCount);
+    lastPageItem.text(this.pagesCount);
+  }
+
+  /**
+   * @returns {Number}
+   *
+   * @private
+   */
+  getLimit() {
+    return this.$paginationContainer.find(this.selectorsMap.limitSelect).val();
+  }
+
+  /**
+   *
+   * @param page
+   *
+   * @returns {Number}
+   */
+  validatePageNumber(page) {
```

Comment:
```suggestion
  getValidatedPageNumber(page) {
```

A method starting with `validateSomething` would be expected to throw an error or return a bool, not modify the input